lib/board.rb

```diff
@@ -0,0 +1,82 @@
+require('pry')
+
+class Board
+  attr_accessor :cells
+
+  WIN_COMBINATIONS = [
+    [0,1,2],
+    [3,4,5],
+    [6,7,8],
+    [0,3,6],
+    [1,4,7],
+    [2,5,8],
+    [0,4,8],
+    [2,4,6]
+  ]
+
+  def initialize
+    reset!
+  end
+
+  def won?
+    WIN_COMBINATIONS.each do |win|
+      if win.all? { |ind| cells[ind] =="X" } || win.all? { |ind|  cells[ind] =="O" }
+        return win
+      end
+    end
+    false
+  end
+
+  def full?
```

Comment:
Use the most expressive and efficient iterator for a problem. `Board#full?` is asking if all elements are not empty strings. You can use logic that is symmetrical to those semantics with the [`#all?`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-all-3F) iterator.

`cells.all? {|cell| cell != " "}`

This is also more efficient in that the loop will stop after the iteration for which a cell is not empty.